.github/workflows/npm-publish.yml

```diff
@@ -1,29 +1,35 @@
-# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
-# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
-
 name: Publish to Npm
 
-on: push
+on:
+  workflow_run:
+    workflows: ["ESLint", "Typecheck", "Test"]
+    types:
+      - completed
+  push:
+    branches:
+      - master
 
 jobs:
   publish:
     runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
I am not sure, but I think this line is ensuring that all other work flow should success before publishing.

Reply:
exactly you can read more about it in the offical docs:

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow